### PR TITLE
Update Buffer.php

### PR DESCRIPTION
**Pull request by Tatikoma**
It's allows to run other libs based on reactphp in blocking mode with little wrapper, if need e.g. socks client based on react. Also it may be useful for some debugging purposes.

### CODE REVIEW

File: tests/BufferTest.php [hunk 1]
```diff
@@ -34,6 +34,24 @@ public function testWrite()
...
+        $loop = $this->createWriteableLoopMock();
+        
+        $buffer = new Buffer($stream, $loop);
+        $buffer->on('error', $this->expectCallableNever());
+        
+        $this->assertSame(0, $buffer->getBufferSize());
+        
+        $buffer->write("foobar\n");
+        rewind($stream);
+        $this->assertSame(strlen("foobar\n"), $buffer->getBufferSize() + strlen(fread($stream, 1024)));
```
Comment (WyriHaximus) [inline tests/BufferTest.php]: Trying to figure out why you're doing the `fread` here, what is the reason for that?

Comment (Tatikoma) [inline tests/BufferTest.php]: Method $buffer->write() writes data into $buffer->data, that push it to stream.
For this reason  size of actual buffer plus size of stream data should be equal to originally written data.

Size of buffer plus size of read data must be equal to originally written data.

**Comment by clue** [comment]
Thanks for the PR!

However, I'm not sure I understand the motivation for these changes. Can you give a use-case / example on how this is going to be used?

Thanks!

> need e.g. socks client based on react

Not sure if this is related, but I noticed you also forked my [SOCKS project](https://github.com/clue/php-socks). It's currently in maintenance mode (see its README), have you considered using my [reactive SOCKS project](https://github.com/clue/php-socks-react) instead? If this is related, it *might* make this PR obsolete, I suppose?

**Comment by Tatikoma** [comment]
I was written PHP library which works with remote services via fsockopen PHP function. Now i need to add SOCKS-proxy support to my library, but i cant change all projects that uses my code, so library interface must be in blocking mode.

I wasnt found any support of blocking mode in php-socks-react library.

With this PR i written next code to write data into buffer in blocking mode (using php-socks-react library):

$tcp->create('www.google.com',80)->then(function (React\Stream\Stream $stream) {
    echo 'connected to www.google.com:80';
    $stream->write("GET / HTTP/1.0\r\n\r\n");
    while($stream->getBuffer()->getBufferSize()){
         // i really need to block script execution until data 'll be written to socket
         $loop->tick();
        usleep(10000);
    }
});

Whats i doing is large hack, but i wasnt not found any good libs to do SOCKS connection except of php-socks-react, so using this lib in blocking mode looks okay.

File: tests/BufferTest.php (hunk 1 above)
Comment (WyriHaximus) [inline tests/BufferTest.php]: Might have misunderstood the code, but my assumption was that `strlen("foobar\n")` would equal `$buffer->getBufferSize()` as the buffer holds `"foobar\n"`

Comment (Tatikoma) [inline tests/BufferTest.php]: Buffer doesnt hold "foobar\n". In first commit i written like u said and got surprised.

**Comment by clue** [comment]
Closed via #43.

**Pull request closed without merging**